src/sentry/integrations/slack/notify_action.py

```diff
@@ -0,0 +1,141 @@
+
+    def clean_channel(self):
+        channel = self.cleaned_data.get('channel', '').lstrip('#')
+        team = self.cleaned_data.get('team')
+
+        channel_id = self.channel_transformer(team, channel)
+
+        if channel_id is None and team is not None:
+            params = {
+                'channel': channel,
+                'team': dict(self.fields['team'].choices).get(int(team)),
+            }
+
+            raise forms.ValidationError(
+                _('The #%(channel)s channel does not exist in the %(team)s Slack team.'),
+                code='invalid',
+                params=params,
+            )
+
+        return channel
+
+
+class SlackNotifyServiceAction(EventAction):
+    form_cls = SlackNotifyServiceForm
+    label = 'Send a notification to the Slack {team} team in {channel}'
+
+    def is_enabled(self):
+        return self.get_integrations().exists()
+
+    def after(self, event, state):
+        channel = self.get_option('channel')
+        try:
+            integration = Integration.objects.get(
+                provider='slack',
+                organizations=self.project.organization,
+            )
+        except Integration.DoesNotExist:
+            self.logger.error('rule.fail.slack_notify.missing_integration')
+            return
+
+        def send_notification(event, futures):
+            attachment = build_attachment(event.group, event=event)
+
+            payload = {
+                'token': integration.metadata['access_token'],
+                'channel': channel,
+                'attachments': json.dumps([attachment]),
+            }
+
+            session = http.build_session()
+            resp = session.post('https://slack.com/api/chat.postMessage', data=payload)
+            resp.raise_for_status()
+            resp = resp.json()
+            if not resp.get('ok'):
+                self.logger.error('rule.fail.slack_post', extra={'error': resp.get('error')})
+
+        metrics.incr('notifications.sent')
+        yield self.future(send_notification)
+
+    def render_label(self):
+        try:
+            integration_name = Integration.objects.get(
+                provider='slack',
+                organizations=self.project.organization,
+                id=self.data['team'],
+            ).name
+        except Integration.DoesNotExist:
+            integration_name = '[removed]'
+
+        return self.label.format(
+            team=integration_name,
+            channel='#' + self.data['channel'],
+        )
+
+    def get_integrations(self):
+        return Integration.objects.filter(
+            provider='slack',
+            organizations=self.project.organization,
+        )
+
+    def get_channel_id(self, integration_id, value):
+        try:
+            integration = Integration.objects.get(
+                provider='slack',
+                organizations=self.project.organization,
+                id=integration_id,
```

Comment:
Is this intended to be the numeric primary key for the column? Or is this supposed to map to the `external_id`?

Reply:
This is intended to be the numeric id of the integration. I'll update value to be channel_name which makes this signature a little more clear I think.

Reply:
d05cfffb6f4c2aa2bd919e89ff7e8f9e148a2e01

Reply:
https://github.com/getsentry/sentry/pull/7050/commits/03916fead2a2c666630084aba65ceee6811d3ac5 this actually.